skbio/alignment/_tabular_msa.py

```diff
@@ -2234,8 +2244,17 @@ def join(self, other, how='strict'):
                 [self.positional_metadata, other.positional_metadata],
                 ignore_index=True, **concat_kwargs)
 
+            if not self.has_positional_metadata():
+                del self.positional_metadata
```

Comment:
Talked offline with @jairideout, we aren't going to support this. 

 1. Users should use `metadata` for this kind of situation. There are so many ways to create properties which may not be propagated, it is the entire reason `metadata` exists.
 2. If we are monkeypatching methods, create a subclass and then post an issue about how the subclass doesn't propogate.
 3. This only happens if it's empty
 4. If after all of the above, it is still done, users get an AttributeError which is noisy and easy to detect.